components/arm/fake/fake.go

```diff
@@ -125,6 +125,27 @@ type Arm struct {
 	model      referenceframe.Model
 }
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (a *Arm) UpdateAction(c *config.Component) config.UpdateActionType {
+	if newCfg, ok := c.ConvertedAttributes.(*AttrConfig); ok {
```

Comment:
don't we also need to check the ArmModel? Probably is useful to refactor in such a way that the internals of the constructor can be reused here to modify the existing arm so that we don't have to repeat ourselves with writing the conditional logic to parse the config

Reply:
sgtm